### PR TITLE
[mlir][transform] Allow passing various library files to interpreter.

### PR DESCRIPTION
**Pull request by ingomueller-net**
The transfrom interpreter accepts an argument to a "library" file with named sequences. This patch exteneds this functionality such that (1) several such individual files are accepted and (2) folders can be passed in, in which all `*.mlir` files are loaded.

This PR depends on and therefor currently includes ~~#67241~~ #67560.

### CODE REVIEW

**Comment by ingomueller-net** [comment]
@qcolombet: This just got a massive feature boost and should be relatively usable now.

File: mlir/lib/Dialect/Transform/Transforms/TransformInterpreterPassBase.cpp [hunk 1]
```diff
@@ -611,53 +613,131 @@ LogicalResult transform::detail::interpreterBaseRunOnOperationImpl(
...
+      return emitError(loc) << "failed to verify transform module";
+  }
 
-  if (parsedTransformModule) {
-    sharedTransformModule = std::make_shared<OwningOpRef<ModuleOp>>(
-        std::move(parsedTransformModule));
+  // Assemble list of library files.
+  SmallVector<std::string> libraryFileNames;
+  for (const std::string &path : transformLibraryPaths) {
+    auto loc = FileLineColLoc::get(context, transformFileName, 0, 0);
```
Comment (ftynse) [inline mlir/lib/Dialect/Transform/Transforms/TransformInterpreterPassBase.cpp:637]: transformFileName -> path

File: mlir/lib/Dialect/Transform/Transforms/TransformInterpreterPassBase.cpp [hunk 2]
```diff
@@ -611,53 +613,131 @@ LogicalResult transform::detail::interpreterBaseRunOnOperationImpl(
...
+      }
+
+      LLVM_DEBUG(DBGS() << "  Adding '" << fileName << "' to list of files\n");
+      libraryFileNames.push_back(fileName);
+    }
+
+    if (ec)
+      return emitError(loc) << "error while opening files in '" << path
+                            << "': " << ec.message();
+  }
```
Comment (ftynse) [inline mlir/lib/Dialect/Transform/Transforms/TransformInterpreterPassBase.cpp:676]: Could we refactor this expansion into a static function that we just call from here?

**Review by ftynse — COMMENTED** [review]
LGTM

**Review by ingomueller-net — COMMENTED** [review]
Thanks for the review! Awesome, this is finally getting to the finish line :) Will merge when CI has passed.

**Pull request merged**